content/docs/engine-dev/architecture/prefabs/_index.md

```diff
@@ -0,0 +1,648 @@
+
+
+
+
+## Overview
+<!-- <a name="overview"> -->
+
+To get an understanding of what prefabs are, we first need to understand what entities are. Entities are the foundational game objects that encapsulate logic required to build game worlds. An entity is made up of components that define its properties. More information about entities and components can be found in [Guide to Component Development](/docs/user-guide/programming/components). 
+
+Prefabs serve the purpose of easily building and modifying large and complex worlds for games. Creating large worlds often requires duplicating a lot of entities in the world. While this can be done by simply creating a bunch of entities and duplicating them in the editor, modifying them becomes a pain point since you would have to go into each of those entities and make the same changes. You would also have to come up with custom tracking mechanisms to identify which entities to modify in which areas of the world. In addition, you may also want some entities to retain their old properties and not to change along with other entities. All of these problems are resolved by prefabs.
+
+Simply put, a prefab is a collection of entities and nested prefabs. The further sections will explain how this collection of entities and nested prefabs are used to solve the above problems and many more.
+
+
+### Prefab file format
+<!-- <a name="prefab-file-format"> -->
+
+Prefab files use JSON format. A prefab file has the following fields.
+
+Container Entity  
+: A wrapper entity, which is the parent of all other entities and nested prefabs contained within a prefab. This is also used to distinguish prefabs from regular entities in the <a href="/docs/user-guide/editor/entity-outliner/">Entity Outliner</a>. A prefab must have a single container entity.
+
+Entities
+: A container for entities owned by a prefab. A prefab may contain zero or more nested entities.
+
+Instances
+: A container for nested prefabs owned by a prefab. A prefab may contain zero or more nested prefabs.
+
+Source
+: The relative path to the prefab file that the nested prefab needs to load from.
+
+Patches
+: A list of changes that get applied on top of the nested prefab.
+
+For example, this is a prefab that contains a single entity and a single nested prefab:
+
+<details>
+<summary>Car.prefab</summary>
+
+```json
+{
+    "ContainerEntity": {
+        "Id": "ContainerEntity",
+        "Name": "Car",
+        "Components": {
+            "Component_[12748530028122391598]": {
+                "$type": "SelectionComponent",
+                "Id": 12748530028122391598
+            },
+            "Component_[13392385766383931996]": {
+                "$type": "EditorOnlyEntityComponent",
+                "Id": 13392385766383931996
+            },
+            "Component_[14257310886211380255]": {
+                "$type": "EditorInspectorComponent",
+                "Id": 14257310886211380255
+            },
```

Comment:
should we simplify this to omit unnecessary components for the purpose of explaining?

Reply:
The content is collapsed by default. So, it won't be a distraction. I prefer that it be as close to the real file as possible since we are showing this by saying, "This is what a prefab file looks like". It might be confusing to users why their prefab looks much bigger than the one shown in documentation.